internal/services/recoveryservices/recovery_services_vault_resource.go

```diff
@@ -249,7 +249,17 @@ func resourceRecoveryServicesVaultCreate(d *pluginsdk.ResourceData, meta interfa
 		vault.Sku.Tier = utils.String("Standard")
 	}
 
+	requireAddtionalUpdate := false
+	updatePatch := vaults.PatchVault{
+		Properties: &vaults.VaultProperties{},
+	}
+	// The `immutability` could only be set to `Locked` from `Unlocked`
 	if immutability, ok := d.GetOk("immutability"); ok {
+		if immutability == string(vaults.ImmutabilityStateLocked) {
```

Comment:
Can we describe the intent a bit clearer, e.g.:
```suggestion
	if immutability, ok := d.GetOk("immutability"); ok {
	        // The API doesn't allow to set the immutability to "Locked" on ceartion.
	        // Here we firstly make it "Unlocked", and once created, we will update it to "Locked".
	        // Note: The `immutability` could be transitioned only from `Unlocked` to `Locked`.
		if immutability == string(vaults.ImmutabilityStateLocked) {
```